### PR TITLE
mypy: fix type checking error

### PR DESCRIPTION
**Pull request by davidculley**
To have an easier time in my other pull requests, when figuring out which type errors were introduced by me and which already existed, I'm fixing some of the type errors on the current main branch.

This pull request fixes this mypy error:

> error: Argument 1 to "_answerCard" of "Reviewer" has incompatible type "int"; expected "Literal[1, 2, 3, 4]"  [arg-type]

The function `_answerCard` wants only numbers 1, 2, 3 or 4 (that is, a variable of type `Literal[1, 2, 3, 4]`. But we're passing a variable of type `int` which can hold many more numbers than these mentioned four.

I'm basically just

- casting `int` to `Literal[1, 2, 3, 4]` (which by itself would already suffice)
- adding a check that the given number is indeed one of the four numbers, so that we don't mute mypy only to miss runtime errors when there actually is one because mypy stayed silent since we muted it.

### CODE REVIEW

File: qt/aqt/reviewer.py [hunk 1]
```diff
@@ -591,6 +591,20 @@ def korean_shortcuts(
     def _shortcutKeys(
         self,
     ) -> Sequence[tuple[str, Callable] | tuple[Qt.Key, Callable]]:
+
+        def generate_default_answer_keys():
+            for ease in aqt.mw.pm.default_answer_keys:
+                key = aqt.mw.pm.get_answer_key(ease)
+                if key:
```
Comment (dae) [inline qt/aqt/reviewer.py:598]: Walrus operator would make this more concise

File: qt/aqt/reviewer.py [hunk 2]
```diff
@@ -591,6 +591,20 @@ def korean_shortcuts(
...
+                    if ease in (1, 2, 3, 4):
```
Comment (dae) [inline qt/aqt/reviewer.py:599]: We control default_answer_keys, so there should be no need to check this.

File: qt/aqt/reviewer.py (hunk 1 above)
Comment (davidculley) [reply qt/aqt/reviewer.py:598]: Should `key` only be not `None` or does it also need to be different from the empty string?

In other words: I would like to change `if key:` to `if key is not None:`, provided this doesn't change its meaning.

File: qt/aqt/reviewer.py (hunk 2 above)
Comment (davidculley) [reply qt/aqt/reviewer.py:599]: I'll remove the check then. 👍

File: qt/aqt/reviewer.py (hunk 1 above)
Comment (dae) [reply qt/aqt/reviewer.py]: The user can enter an empty string in the preferences, so we also need to guard against that.

**Comment by dae** [comment]
🚀

File: qt/aqt/reviewer.py (hunk 1 above)
Comment (davidculley) [reply qt/aqt/reviewer.py]: Thank you for fixing the code. I wasn't sure about that.

File: qt/aqt/reviewer.py [hunk 3]
```diff
@@ -591,6 +591,18 @@ def korean_shortcuts(
     def _shortcutKeys(
         self,
     ) -> Sequence[tuple[str, Callable] | tuple[Qt.Key, Callable]]:
+
+        def generate_default_answer_keys() -> (
+            Generator[tuple[str, partial], None, None]
+        ):
+            for ease in aqt.mw.pm.default_answer_keys:
+                key = aqt.mw.pm.get_answer_key(ease)
+                if key is None:
```
[the remark this answers is not in the document]
Comment (davidculley) [reply qt/aqt/reviewer.py]: Thank you.

**Pull request merged**